libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/NavigationV2Router.kt

```diff
@@ -0,0 +1,42 @@
+package com.mapbox.navigation.base.internal
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.navigation.base.CurrentIndicesSnapshot
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.route.NavigationRouter
+import com.mapbox.navigation.base.route.NavigationRouterRefreshCallback
+
+/**
+ * Extends [NavigationRouter] to also provide ability for refreshing routes partially
+ * using indices stored in [CurrentIndicesSnapshot].
+ */
+interface NavigationV2Router : NavigationRouter {
```

Comment:
```suggestion
interface NavigationRouterV2 : NavigationRouter {
```
This would honestly be my preference, but if you have a strong opinion, we can leave as is.